FreydCategoriesForCAP/gap/AdditiveClosure.gi

```diff
@@ -1356,6 +1362,256 @@ InstallGlobalFunction( INSTALL_FUNCTIONS_FOR_ADDITIVE_CLOSURE,
         
     fi;
     
+    if HasIsMonoidalCategory( underlying_category ) and IsMonoidalCategory( underlying_category ) then
+        
+        ##
+        AddTensorUnit( category,
+          function( cat )
+            
+            return ObjectConstructor( cat, [ TensorUnit ( UnderlyingCategory( cat ) ) ] );
+            
+          end );
+        
+        ##
+        AddTensorProductOnObjects( category,
+          function( cat, obj_1, obj_2 )
+            local underlying_category, objdat_1, objdat_2, len_1, len_2;
+            
+            underlying_category := UnderlyingCategory( cat );
+            objdat_1 := ObjectDatum( cat, obj_1 );
+            objdat_2 := ObjectDatum( cat, obj_2 );
+            len_1 := Length( objdat_1 );
+            len_2 := Length( objdat_2 );
+            
+            return ObjectConstructor( cat,
+                    List( [ 0 .. len_1 * len_2 - 1 ], i ->
+                      TensorProductOnObjects( underlying_category,
+                        objdat_1[1 + QuoInt( i, len_2 )],
+                        objdat_2[1 + RemInt( i, len_2 )] ) ) );
+            
+          end );
+        
+        ##
+        AddTensorProductOnMorphismsWithGivenTensorProducts( category,
```

Comment:
If the underlying category is just a commutative ring, does this implementation match
1. $M \otimes N$ where $\otimes$ is the Kronecker product of matrices,
2. $(N^T \otimes M^T)^T$,
3. neither?

Background of the question: It would be nice if the implementation here would coincide with the existing implementation in CategoryOfRows.

Reply:
In order to coincide with the current implementation of CategoryOfRows the monoidal structure applied to RingAsCategory yield the Kronecker product, so 1.

Reply:
Perfect! Yes, an example demonstrating that the implementation yields exactly the Kronecker product would be nice.

Reply:
done